torch/csrc/autograd/variable.h

```diff
@@ -623,16 +640,25 @@ inline Variable Variable::detach() const {
 }
 
 inline void Variable::set_gradient_edge(Edge edge) noexcept {
+  materialize_autograd_meta();
   get_autograd_meta()->grad_fn_ = std::move(edge.function);
   get_autograd_meta()->output_nr_ = edge.input_nr;
 }
 
 inline uint32_t Variable::output_nr() const noexcept {
-  return get_autograd_meta()->output_nr_;
+  if (get_autograd_meta()) {
+    return get_autograd_meta()->output_nr_;
+  } else {
+    return 0;
```

Comment:
Have you considered this to be an error? This should never be called on a Tensor or default constructed autograd meta no?

Reply:
An error might be appropriate. But for this diff, I opted for the semantics preserving option, which is zero (the default value).